snuba/migrations/table_engines.py

```diff
@@ -88,6 +88,14 @@ def _get_engine_type(self, cluster: ClickhouseCluster, table_name: str) -> str:
             return f"ReplicatedReplacingMergeTree('/clickhouse/tables/{{layer}}-{{shard}})/{table_name}', '{{replica}}', {self.__version_column})"
 
 
+class SummingMergeTree(MergeTree):
+    def _get_engine_type(self, cluster: ClickhouseCluster, table_name: str) -> str:
+        if cluster.is_single_node():
+            return f"SummingMergeTree()"
+        else:
+            return f"ReplicatedSummingMergeTree('/clickhouse/tables/{{layer}}-{{shard}})/{table_name}', '{{replica}}')"
```

Comment:
Have you ever tried to execute locally the distributed migrations to ensure they (local + distributed) generate a working database ?

Reply:
I haven't, anything that's not single node is untested. I'd like to start running this in our test suite at some point.